packages/workflow/src/RoutingNode.ts

```diff
@@ -564,6 +569,102 @@ export class RoutingNode {
 						responseData.push(...tempResponseData);
 					} while (tempResponseData.length && tempResponseData.length === properties.pageSize);
 				}
+				// Generic pagination, using the contents of the previous response to
+				// decide whether a new one should be made.
+				else if (requestOperations.pagination.type === 'generic') {
+					const { properties } = requestOperations.pagination;
+
+					// Merge request properties coming in from different parts of the declaration
+					// (avoiding Object.assign() as it only does a shallow merge)
+					if (properties.request) {
+						requestData.options = merge({}, requestData.options, properties.request);
+					}
+
+					let previousResponseData: INodeExecutionData | null = null;
+					let currentResponseData: INodeExecutionData[] = [];
+					let canContinue = false;
+
+					do {
+						// Deep-clone requestData, as evaluating a parameter would overwrite it
+						const currentRequestData: DeclarativeRestApiSettings.ResultOptions = JSON.parse(
+							JSON.stringify(requestData),
+						);
+
+						// Evaluate any additional qs/body/headers parameters into the request
+						type QueryOption = keyof DeclarativeRestApiSettings.HttpRequestOptions;
+						const configurableQueryOptions: QueryOption[] = ['qs', 'body', 'headers'];
```

Comment:
This now supports changing any of query string, body and headers based on the response values. I'm thinking just supporting the query string would likely cover almost all the uses cases anyway, so should we consider dropping body and header?

Reply:
I think it makes sense to keep those as options; there are no drawbacks and it's flexible.